src/com/epam/javacourse/homework4/GreenhouseManagement.java

```diff
@@ -0,0 +1,116 @@
+package com.epam.javacourse.homework4;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class GreenhouseManagement implements Greenhouse {
+    private final List<Plant> plants;
+    private final List<Plant> availablePlants;
+    public double temperature;
+
+    public GreenhouseManagement() {
+        this.plants = new ArrayList<>();
+        this.availablePlants = new ArrayList<>();
+        this.temperature = 0;
+    }
+
+    public void addPlant(Plant plant) {
+        if (!plants.contains(plant) && !availablePlants.contains(plant)) {
+            this.availablePlants.add(plant);
+            System.out.println("Plant " + plant.getPlantName() + " added to the list of available plants.");
+        } else {
+            System.out.println("Plant " + plant.getPlantName() + " is already in the greenhouse.");
+        }
+    }
+
+    @Override
+    public void buyPlant(String plantName) {
+        Plant plantToBuy = null;
+        for (Plant plant : availablePlants) {
+            if (plant.getPlantName().equals(plantName)) {
+                plantToBuy = plant;
+                break;
+            }
+        }
+        if (plantToBuy == null) {
+            System.out.println("No plant with the given name found in the list of available plants.");
+        } else {
+            availablePlants.remove(plantToBuy);
+            plants.add(plantToBuy);
+            System.out.println("Plant " + plantName + " is bought and added to the greenhouse.");
+        }
+    }
+
+    /**
+     * An anonymous inner class that represents a watering system for the greenhouse.
+     */
+    public abstract static class WateringSystem {
```

Comment:
It's not an anonymous class. Please, take a look at the official Java documentation https://docs.oracle.com/javase/tutorial/java/javaOO/anonymousclasses.html#syntax-of-anonymous-classes